gobblin-core/src/main/java/gobblin/writer/AvroDataWriterBuilder.java

```diff
@@ -32,7 +32,11 @@
  * @author ynli
  */
 @SuppressWarnings("unused")
-public class AvroDataWriterBuilder extends DataWriterBuilder<Schema, GenericRecord> {
+public class AvroDataWriterBuilder extends PartitionAwareDataWriterBuilder<Schema, GenericRecord> {
+
+  @Override public boolean validatePartitionSchema(Schema partitionSchema) {
```

Comment:
You can make this a default implementation in `PartitionAwareDataWriterBuilder`?